streams/src/main/java/org/apache/kafka/streams/kstream/internals/suppress/SuppressedInternal.java

```diff
@@ -62,22 +74,25 @@ public boolean equals(final Object o) {
         if (this == o) return true;
         if (o == null || getClass() != o.getClass()) return false;
         final SuppressedInternal<?> that = (SuppressedInternal<?>) o;
-        return Objects.equals(bufferConfig, that.bufferConfig) &&
-            Objects.equals(getTimeToWaitForMoreEvents(), that.getTimeToWaitForMoreEvents()) &&
-            Objects.equals(getTimeDefinition(), that.getTimeDefinition());
+        return suppressTombstones == that.suppressTombstones &&
+            Objects.equals(name(), that.name()) &&
```

Comment:
nit: why `name()` instead of `name` ? Similar other parameters; also in `hashCode()`.

Reply:
My code generator in IDEA defaults to using the getter if available. I assume to account for comparing this with an instance of a subclass.

Do you prefer using the field? I can swap them out.

Reply:
Oh, weird. It only happened in this class. I guess IDEA only thinks it's a getter if it follows the bean spec getter name (like `getBufferConfig`). I fixed it.